packages/snaps-jest/src/internals/simulation/interface.ts

```diff
@@ -100,37 +119,271 @@ function resolveWithInput(runSaga: RunSagaFunction) {
 }
 
 /**
- * Get a user interface object from a Snap.
+ * Get the stored user interface from the store.
  *
- * @param runSaga - A function to run a saga outside the usual Redux flow.
- * @param snapId - The Snap ID.
  * @param controllerMessenger - The controller messenger used to call actions.
+ * @param snapId - The Snap ID.
  * @yields Takes the set interface action.
  * @returns The user interface object.
  */
-export function* getInterface(
-  runSaga: RunSagaFunction,
-  snapId: SnapId,
+function* getStoredInterface(
   controllerMessenger: RootControllerMessenger,
-): SagaIterator<SnapInterface> {
+  snapId: SnapId,
+): SagaIterator<Interface & { content: Component }> {
   const currentInterface: Interface | null = yield select(getCurrentInterface);
+
   if (currentInterface) {
     const { content } = controllerMessenger.call(
       'SnapInterfaceController:getInterface',
       snapId,
       currentInterface.id,
     );
-    return getInterfaceResponse(runSaga, currentInterface.type, content);
+
+    return { ...currentInterface, content };
   }
 
   const { payload }: PayloadAction<Interface> = yield take(setInterface.type);
-  const { type, id } = payload;
 
   const { content } = controllerMessenger.call(
     'SnapInterfaceController:getInterface',
     snapId,
+    payload.id,
+  );
+
+  return { ...payload, content };
+}
+
+/**
+ * Get a Button or an Input from an interface.
+ *
+ * @param content - The interface content.
+ * @param name - The element name.
+ * @returns An object containing the element and the form name if it's contained in a form, otherwise undefined.
+ */
+export function getElement(
+  content: Component,
+  name: string,
+):
+  | {
+      element: Button | Input;
+      form?: string;
+    }
+  | undefined {
+  const { type } = content;
+
+  if (
+    (type === NodeType.Button || type === NodeType.Input) &&
+    content.name === name
+  ) {
+    return { element: content };
+  }
+
+  if (type === NodeType.Form) {
+    for (const element of content.children) {
+      const result = getElement(element, name);
+      if (result) {
+        return { element: result.element, form: content.name };
+      }
+    }
+  }
+
+  if (type === NodeType.Panel) {
+    for (const element of content.children) {
+      const result = getElement(element, name);
+
+      if (result) {
+        return result;
+      }
+    }
+  }
```

Comment:
Maybe you can simplify this by checking if the node has a `children` property, and returning `{ element: ..., name: node?.name }`?